rcl/test/rcl/test_events.cpp

```diff
@@ -140,14 +130,26 @@ class CLASSNAME (TestEventFixture, RMW_IMPLEMENTATION) : public ::testing::Test
     ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
 
     // init subscription
-    ret = setup_subscriber(deadline, lifespan, lease_duration, liveliness_policy);
+    ret = setup_subscriber(sub_qos_profile);
     ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
 
     // init subscription event
     subscription_event = rcl_get_zero_initialized_event();
     ret = rcl_subscription_event_init(&subscription_event, &subscription, sub_event_type);
     ASSERT_EQ(ret, RCL_RET_OK) << rcl_get_error_string().str;
+  }
+
+  void setup_publisher_and_subscriber_and_assert_discovery(
```

Comment:
This was the earlier `setup_publisher_and_subscriber` function which did two things:
-  setup publisher and subscription
- asserted that the publisher and subscription discovered each other

I have renamed this function and also moved the "setup publisher and subscription" portion of the code into a new function `setup_publisher_and_subscriber`. 

All existing tests have been modified to call this function.